tests/end_to_end/end_to_end.cpp

```diff
@@ -1064,4 +1082,351 @@ TEST_CASE("link_tests", "[end_to_end]")
     REQUIRE(hook.attach(program) == EBPF_INVALID_ARGUMENT);
 
     hook.detach();
-}
\ No newline at end of file
+}
+
+TEST_CASE("map_reuse", "[end_to_end]")
+{
+    _test_helper_end_to_end test_helper;
+    single_instance_hook_t hook(EBPF_PROGRAM_TYPE_XDP, EBPF_ATTACH_TYPE_XDP);
+    program_info_provider_t xdp_program_info(EBPF_PROGRAM_TYPE_XDP);
+
+    // First create and pin the maps manually.
+    int inner_map_fd = bpf_create_map(BPF_MAP_TYPE_ARRAY, sizeof(__u32), sizeof(__u32), 1, 0);
+    REQUIRE(inner_map_fd > 0);
+
+    int outer_map_fd = bpf_create_map_in_map(BPF_MAP_TYPE_HASH_OF_MAPS, nullptr, sizeof(__u32), inner_map_fd, 1, 0);
+    REQUIRE(outer_map_fd > 0);
+
+    // Verify we can insert the inner map into the outer map.
+    __u32 outer_key = 0;
+    int error = bpf_map_update_elem(outer_map_fd, &outer_key, &inner_map_fd, 0);
+    REQUIRE(error == 0);
+
+    // Pin the outer map
```

Comment:
```suggestion
    // Pin the outer map.
```
nit for consistency with the other comments